app/src/hooks/useStakingOptions.ts

```diff
@@ -73,13 +73,22 @@ async function fetchData(provider: AnchorProvider): Promise<SoParams[]> {
         continue;
       }
 
+      const quote = quoteMint.toString();
+      const isUpside =
+        quote === Config.usdcMintPk().toString() ||
+        quote === Config.usdhMintPk().toString() ||
+        quote === Config.usdtMintPk().toString();
       const soParams = {
         key: `${soName}-${soMint.toString()}`,
         name: soName,
         authority: new PublicKey(authority),
         expiration: new Date(Number(optionExpiration) * 1_000).toDateString().split(' ').slice(1).join(' '),
         expirationInt: Number(optionExpiration),
-        strike: roundedStrike,
+        strike: calculateStrikeAtomsPerBaseToken({
+          isUpside,
+          mint: isUpside ? quoteMint : baseMint,
+          strike: strike.toNumber(),
+        }).toString(),
```

Comment:
why a string? seems like it is something that should be a number.
or are we trimming to correct decimals here so we dont format later?